src/accessors/IModify.ts

```diff
@@ -0,0 +1,241 @@
+     */
+    setSender(sender: IUser): IMessageBuilder;
+
+    /**
+     * Sets the text of the message.
+     *
+     * @param text the actual text
+     */
+    setText(text: string): IMessageBuilder;
+
+    /**
+     * Sets the emoji to use for the avatar, this overwrites the current avatar
+     * whether it be the user's or the avatar url provided.
+     *
+     * @param emoji the emoji code
+     */
+    setEmojiAvatar(emoji: string): IMessageBuilder;
+
+    /**
+     * Sets the url which to display for the avatar, this overwrites the current
+     * avatar whether it be the user's or an emoji one.
+     *
+     * @param avatarUrl image url to use as the avatar
+     */
+    setAvatarUrl(avatarUrl: string): IMessageBuilder;
+
+    /**
+     * Sets the display text of the sender's username that is visible.
+     *
+     * @param alias the username alias to display
+     */
+    setUsernameAlias(alias: string): IMessageBuilder;
+
+    /**
+     * Adds one attachment to the message's list of attachments, this will not
+     * overwrite any existing ones but just adds.
+     *
+     * @param attachment the attachment to add
+     */
+    addAttachment(attachment: IMessageAttachment): IMessageBuilder;
+
+    /**
+     * Sets the attachments for the message, replacing and destroying all of the current attachments.
+     *
+     * @param attachments array of the attachments
+     */
+    setAttachments(attachments: Array<IMessageAttachment>): IMessageBuilder;
+
+    /**
+     * Replaces an attachment at the given position (index).
+     * If there is no attachment at that position, there will be an error thrown.
+     *
+     * @param position the index of the attachment to replace
+     * @param attachment the attachment to replace with
+     */
+    replaceAttachment(position: number, attachment: IMessageAttachment): IMessageBuilder;
+
+    /**
+     * Removes an attachment at the given position (index).
+     * If there is no attachment at that position, there will be an error thrown.
+     *
+     * @param position the index of the attachment to remove
+     */
+    removeAttachment(position: number): IMessageBuilder;
+
+    /**
+     * Sets the user who is updating this message.
+     * This is required if you are modifying an existing message.
+     *
+     * @param user the updater
+     */
+    setUpdater(user: IUser): IMessageBuilder;
+
+    /**
+     * Gets the resulting message that has been built up to the point of calling it.
+     * Note: modifying the returned value will have no effect.
+     */
+    getMessage(): IMessage;
+
+    /**
+     * Finishes the building and will send/update the message, unless told otherwise.
+     * Please note, that a room and sender must be associated otherwise this
+     * will fail and throw an error.
+     *
+     * @param preventSending (optional) only build the message and don't send/update it
+     */
+    finish(preventSending?: boolean): IMessage;
```

Comment:
I thought this `finish` represented some kind of `flush` for the modifications buffered till then. So is each step in the builder atomic? In other words: when is the update on the mongo triggered?